runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java

```diff
@@ -566,6 +566,28 @@ public void show()
 							clientBounds.x, clientBounds.y, clientBounds.width, clientBounds.height);
 						// Reset the position, but not the size
 						frame.setLocationRelativeTo(frame.getOwner());
+
+						// Check if the window is now in bounds
+						gc = findDisplayFromBounds(frame.getBounds());
+						if (gc == null)
+						{
+
+							gc = findDisplayFromIntersect(frame.getBounds());
+							if (gc != null)
+							{
+								Rectangle adjustedBounds = gc.getBounds();
+
+								// Adjust for macOS top bar
+								if (OSType.getOSType() == OSType.MacOS)
```

Comment:
I am not really a fan of hard coding the menu bar offsets, have you tried using the screen insets instead? Such as:

```java
var insets = Toolkit.getDefaultToolkit().getScreenInsets(gc);
bounds.y += insets.top;
bounds.height -= (insets.bottom + insets.top);
```

Also maybe offset x too. Or alternatively, I would not be opposed to just resetting the client size to the safe mode size.